### ISSUE TITLE
Unexpected behavior using .populate()

### ISSUE THREAD

**Issue by janek-bieser** [mccormicka/Mockgoose#85]
Using: Mockgoose@1.10.2, Mongoose@3.8.15

Given the following mongoose schema and model
```js
var UserSchema = new mongoose.Schema({
    name: String,
    friends: [{ type: ObjectId, ref: 'User'}]
});
var User = mongoose.model('User', UserSchema);
```
lets pretend the following data exists for the `User` model. The `_id` values are just for illustration, they are supposed to be real ObjectId's
```
User 1: { _id: ObjectId_1, name: 'User 1', friends: [] }
User 2: { _id: ObjectId_2, name: 'User 2', friends: [] }
User 3: { _id: ObjectId_3, name: 'User 3', friends: [ ObjectId_1, ObjectId_2 ] }
```
The following example uses Mongoose only (no Mockgoose)
```js
User.findOne({name: 'User 3'}).populate('friends').exec(function(err, user) {
    // user.name is 'User 3'
    // user.friends contains two User objects (as expected)
    User.findOne({name: 'User 3'}, function(err, user) {
        // user.name is 'User 3'
        // user.friends contains 2 ObjectId's (as I would expect)
    });
});
```
The following example uses Mockgoose
```js
...
mockgoose(mongoose); // I do this at the very beginning
...
User.findOne({name: 'User 3'}).populate('friends').exec(function(err, user) {
    // user.friends contains two User objects (as expected)
    User.findOne({name: 'User 3'}, function(err, user) {
        // user.friends ALSO contains 2 User objects. I would expect 2 ObjectId's
    });
});
```
Not sure if I am doing something wrong here. Feedback would be appreciated.

**Comment by mypark** [comment]
HI - any updates on the open pull request?  I wanted to get the populate() function working on my unit tests and would be great if this solved the populate issue.

**Comment by mypark** [comment]
@yads so it looks like it works if you don't pass any select fields to the populate, but if you do it runs into the same issue .. e.g.  .populate('createdBy', {fname:1, lname:1});

**Comment by ovaris** [comment]
It would be really really cool to have this fixed!

**Comment by winfinit** [comment]
pull request was merged and new 2.0.2 version in now on npm, thank you for your change.
